tests/codestyle/CodestyleTest.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+use Composer\Console\Application;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Console\Input\ArrayInput;
+
+class CodestyleTest extends TestCase
+{
+    protected const FIXTURES = [
+        "singleQuotes",
+        "strictTypes",
+        "unionTypes",
+    ];
+
+    protected Application $application;
+
+    protected function setUp(): void
+    {
+        $this->bootstrapComposer();
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function testFixtures(): void
+    {
+        $this->clearTempDirectory();
+
+        foreach (static::FIXTURES as $fixture) {
+            $this->testFixture($fixture);
+        }
+
+        $this->clearTempDirectory();
+    }
+
+    protected function bootstrapComposer(): void
+    {
+        $this->application = new Application();
+        $this->application->setAutoExit(false);
+    }
+
+    /**
+     * @throws Exception
+     */
+    protected function runComposerEcsCommand(bool $fix = false): bool
+    {
+        $command = $fix ? "ecsf-tmp" : "ecs-tmp";
+        $input = [
+            "command" => $command,
+        ];
+
+        return $this->application->run(new ArrayInput($input)) === 0;
```

Comment:
I think we have 2 options here with regards to the composer script errors returned. We can either ignore them, since it all works anyway or, we can run composer from shell instead of php. 

If we go with the first one, I'd just change a couple of things:
```suggestion
        $command = $fix ? "ecsf-tmp" : "ecs-tmp";

        return $this->composer->run(new StringInput($command)) === 0;
```

- rename `$application` to `$composer` so that it's clear what application are we working with
- use `StringInput` instead of `ArrayInput` as it's simpler

Going with the second option, we're running the composer from shell so we can get rid of the `$application` altogether:

```suggestion
        $command = $fix ? "ecsf-tmp" : "ecs-tmp";

        $result = 0;
        $output = null;
        exec("./vendor/bin/composer ".$command, $output, $result);
        return $result === 0;
```